src/send/hooks.ts

```diff
@@ -0,0 +1,226 @@
+  )
+
+  return {
+    mergedRecipients,
+    searchQuery,
+    setSearchQuery: throttledSearch,
+  }
+}
+
+/**
+ * Fetches recipients based off the search query by fetching from the resolveId
+ * endpoint. The search query is debounced before making a network request in order
+ * to prevent excessive network calls.
+ */
+export function useResolvedRecipients(searchQuery: string): Recipient[] {
+  const [debouncedSearchQuery, setDebouncedSearchQuery] = useState(searchQuery)
+  const debounceSearchQuery = debounce((query: string) => {
+    setDebouncedSearchQuery(query)
+  }, TYPING_DEBOUNCE_MILLSECONDS)
+
+  const defaultCountryCode = useSelector(defaultCountryCodeSelector)
+
+  useEffect(() => {
+    const parsedPhoneNumber = parsePhoneNumber(searchQuery, defaultCountryCode ?? undefined)
+
+    if (parsedPhoneNumber) {
+      debounceSearchQuery(parsedPhoneNumber.e164Number)
+    } else {
+      debounceSearchQuery(searchQuery)
+    }
+  }, [searchQuery, defaultCountryCode])
+  const { result: resolveAddressResult } = useAsync(resolveId, [debouncedSearchQuery])
+  const resolutions = resolveAddressResult?.resolutions ?? []
+  return useMapResolutionsToRecipients(searchQuery, resolutions as NameResolution[])
+}
+
+/**
+ * Returns recent and contact recipients from Redux.
+ */
+export function useSendRecipients() {
+  const contactsCache = useSelector(phoneRecipientCacheSelector)
+  const contactRecipients = useMemo(
+    () => sortRecipients(Object.values(contactsCache)),
+    [contactsCache]
+  )
+  const recentRecipients = useSelector((state) => state.send.recentRecipients)
+  return {
+    contactRecipients,
+    recentRecipients,
+  }
+}
+
+/**
+ * Merges all recipient types, including contacts, recents, resolved, and unique, into a single
+ * ordered list.
+ *
+ * Recipients are ordered by the following precedence:
+ *  - Resolved recipients
+ *  - Recent recipients
+ *  - Contact recipients
+ *  - Unique recipient, if present and no other recipients exist
+ *
+ * If there are any duplicated recipients (by phone number or address), they are dedpulicated,
+ * picking the recipient to show based on the precedence listed above.
+ */
+export function mergeRecipients({
+  contactRecipients,
+  recentRecipients,
+  resolvedRecipients,
+  uniqueSearchRecipient,
+}: {
+  contactRecipients: Recipient[]
+  recentRecipients: Recipient[]
+  resolvedRecipients: Recipient[]
+  uniqueSearchRecipient?: Recipient
+}): Recipient[] {
+  const allRecipients: Recipient[] = []
+  allRecipients.push(...resolvedRecipients)
+  allRecipients.push(...recentRecipients)
+  allRecipients.push(...contactRecipients)
+
+  const mergedRecipients: Recipient[] = []
+  for (const potentialRecipient of allRecipients) {
+    if (
+      !mergedRecipients.find(
+        (mergedRecipient) =>
+          mergedRecipient.e164PhoneNumber === potentialRecipient.e164PhoneNumber ||
+          mergedRecipient.address === potentialRecipient.address
```

Comment:
Multiple recipient's can have a undefined address, not on Valora, so this causes only one match to be returned in cases where there is actually more than one match.

Reply:
fixed! thanks a ton for catching this